internal/discord/commands.go

```diff
@@ -98,7 +98,53 @@ func (d *Discord) Ban(s *discordgo.Session, i *discordgo.InteractionCreate) {
 //	user:		User
 //	reason:		string
 func (d *Discord) Unban(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	optionMap := mapOptions(i)
+	logMsg, _ := d.LogCommand(i.Interaction)
+
+	state := &InteractionState{
+		session:     s,
+		interaction: i,
+		logMsg:      logMsg,
+		isFirst:     true,
+	}
+
+	userToUnban := optionMap["user"].UserValue(nil).ID
+
+	// Check if user exists in ban list
+	member, err := d.Session.GuildBan(state.interaction.GuildID, userToUnban)
+	fmt.Printf("%v", member)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not unban user <@%v>", userToUnban)
+		fmt.Printf("%v: %v\n", tempstr, err)
+
+		err = RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+		if err != nil {
+			fmt.Printf("Unable to send ephemeral message: %v\n", err)
+		}
+
+		return
+	}
+
+	// Dm the user regarding the unban
+	// unbanstr := fmt.Sprintf("You are being unbanned from `%v` for the following reason:\n> %v", GuildName)
+	// d.SendDMToUser(state, userToUnban, unbanstr)
+	//
+	// tempstr := fmt.Sprintf("<@%v> has been unbanned", userToUnban)
+	// RespondAndAppendLog(state, tempstr)
+	// d.EditLogMsg(logMsg)
```

Comment:
@clairefleuret the bot can't message the person because they don't share the same server. is DMing the user a requirement?

Reply:
In that case, no. We will have to do it manually. If anyone is requesting an unban, they would need to be in contact with us anyways.